RobustTrees/RobustOCT.py

```diff
@@ -0,0 +1,195 @@
+import copy
+from gurobipy import *
+import math
+import numpy as np
+import pandas as pd
+import sys
+
+class RobustOCT:
+    def __init__(self, X, y, tree, time_limit, q, s, p, l, seed):
+        '''
+        :param data: The training data
+        :param label: Name of the column representing the class label
+        :param tree: Tree object
+        :param time_limit: The given time limit for solving the MIP
+        :param q_values: probability that a feature is certain
+        :param p: probability that a label is certain (rho)
+        :param l: lambda threshold for uncertainty set
+        :param seed: random seed
+        '''
+        # if our data is given as a data frame, keep the feature labels
+        # otherwise, just give numeric variable names
+        # Make X matrix a pd dataframe, but keep y as an array
+        if isinstance(X, pd.Dataframe):
+            self.cat_features = X.columns
+            self.X = X
+        else:
+            self.cat_features = np.arange(0, self.X.shape[1])
+            self.X = pd.Dataframe(X, columns=self.cat_features)
+
+        if isinstance(self.y, [pd.Series, pd.DataFrame]):
+            self.y = y.values
+            self.labels = np.unique(self.y)
+        else:
+            self.y = y
+            self.labels = np.unique(self.y)
+        # datapoints contains the indicies of our training data
+        self.datapoints = np.arange(0, self.X.shape[0])
+        self.tree = tree
```

Comment:
Sorry, I forgot to push my changes for the Strong Trees in which we handle the data conversion before initializing the Gurobi model. The reason for this is because we need to keep the column names for any tree visualizations, etc but if we keep the data in a dataframe, we lose access to the sklearn API

See below for how to restructure this part to match Strong Trees:
* We now use an [extract metadata function](https://github.com/patrickvossler18/decision_tree_estimators/blob/99418904179a87b38ebb565e60be0a706f2b2025/StrongTrees/StrongTree.py#L50) to store column names and label information
* We pass [the column information](https://github.com/patrickvossler18/decision_tree_estimators/blob/99418904179a87b38ebb565e60be0a706f2b2025/StrongTrees/FlowOCT.py#L9) to the RobustOCT class

Let me know if you have any questions about these changes